### PR TITLE
[NVNET] Add driver for nForce-based NICs 

### PR DESCRIPTION
**Pull request by disean**
## Purpose

Support networking on OG Xbox.

JIRA issue: [CORE-15872](https://jira.reactos.org/browse/CORE-15872)

### CODE REVIEW

File: hal/halx86/generic/usage.c [hunk 1]
```diff
@@ -279,6 +279,9 @@ HalpReportResourceUsage(IN PUNICODE_STRING HalName,
     LARGE_INTEGER CurrentSortValue, SortValue;
     DbgPrint("%wZ Detected\n", HalName);
 
+#if defined(SARCH_XBOX)
+    UNREFERENCED_PARAMETER(Port);
+#else
```
Comment (binarymaster) [inline hal/halx86/generic/usage.c:284]: Hm, I wonder if the normal boot mode is affected (by the serial driver). This code path seems to be triggered only when KD is active.

Comment (HBelusca) [reply hal/halx86/generic/usage.c:284]: Whats the purpose of this hal change, by the way?

Comment (binarymaster) [reply hal/halx86/generic/usage.c:284]: @HBelusca it's in commit message

Comment (HBelusca) [reply hal/halx86/generic/usage.c:284]: > @HBelusca it's in commit message

You see, code comments exist for a reason, too.

File: hal/halx86/generic/usage.c [hunk 2]
```diff
@@ -279,6 +279,9 @@ HalpReportResourceUsage(IN PUNICODE_STRING HalName,
     LARGE_INTEGER CurrentSortValue, SortValue;
     DbgPrint("%wZ Detected\n", HalName);
 
+#if defined(SARCH_XBOX)
+    UNREFERENCED_PARAMETER(Port);
```
Comment (HBelusca) [inline hal/halx86/generic/usage.c:283]:
```suggestion
    /* Do not claim interrupt resources for the KD COM port.
     * The actual COM port lacks SERIRQ, IRQ 4 is hardwired to the NIC. */
    UNREFERENCED_PARAMETER(Port);
```

File: drivers/network/dd/nvnet/nvnet.h [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    BOOLEAN MacInUse;
+    BOOLEAN UseSoftwareMacAddress;
+    BOOLEAN UnitSemaphoreAcquired;
+    BOOLEAN GigabitPhy;
+    BOOLEAN FullDuplex;
+    BOOLEAN PrioritySupported;
+    BOOLEAN VlanTaggingSupported;
+    BOOLEAN ChecksumOffload;
+    BOOLEAN ForceSpeedAndDuplex;
+    BOOLEAN ForceFullDuplex;
```
Comment (Extravert-ir) [inline drivers/network/dd/nvnet/nvnet.h:198]: Maybe make that a bit field?

**Review by Extravert-ir — APPROVED** [review]
If the driver works, I'm fine with going on. Please, check it on a real Xbox first (I guess @dreimer1986 may help)

**Comment by DarkFire01** [comment]
PXL_20220109_004755795

Hi! i just wanted to confirm this works on a real xbox tested today (1/8/2022) :)
Thanks Dreimer!

File: drivers/network/dd/nvnet/netnv.inf [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+FTx = "Tx Pause"
+FBoth = "Rx & Tx Pause"
+No = "No"
+Yes = "Yes"
+PP = "Packet Priority"
+VT = "VLAN Tagging"
+VI = "VLAN ID"
+JF = "Jumbo Frame"
+
+NVNET.DeviceDesc = "nVidia nForce Ethernet Controller"
```
Comment (binarymaster) [inline drivers/network/dd/nvnet/netnv.inf:264]: Maybe it's worth to add other supported adapter descriptions here like nForce2, nForce3, etc. (referenced in `nic.c` line 699) But I can do this myself later, not critical :)

**Review by binarymaster — COMMENTED** [review]
The second commit mentions CORE-15872 twice:

image

File: drivers/network/dd/nvnet/netnv.inf [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+FRx = "Кадры паузы приема"
+FTx = "Кадры паузы передачи"
+FBoth = "Кадры паузы приема и передачи"
+No = "Нет"
+Yes = "Да"
+PP = "Приоритет кадров"
+VT = "Маркировка кадров"
+JF = "Кадры большого размера"
+
+NVNET.DeviceDesc = "nVidia nForce сетевой контроллер"
```
Comment (binarymaster) [inline drivers/network/dd/nvnet/netnv.inf:288]: For the consistency with existing network adater INF files:
```suggestion
NVNET.DeviceDesc = "nVidia nForce PCI сетевой адаптер"
```

**Review by binarymaster — COMMENTED** [review]
Suggestion for last commit message:
```diff
-All NICs are x86 and amd64 devices only.
+All NICs are compatible with x86 and amd64 devices only.
```

File: drivers/network/dd/nvnet/netnv.inf (hunk 5 above)
Comment (disean) [reply drivers/network/dd/nvnet/netnv.inf:288]: This makes sense in general, but nvnet controller is not a card that plugs into a slot

File: drivers/network/dd/nvnet/netnv.inf (hunk 4 above)
Comment (disean) [reply drivers/network/dd/nvnet/netnv.inf:264]: >other supported adapter descriptions

It seems that the word 'nForce' always shows up in the device manager page if I install the vendor's NIC driver

**Pull request merged**